lib/omnibus/packagers/deb.rb

```diff
@@ -395,10 +420,10 @@ def safe_version
     # @return [String]
     #
     def safe_architecture
-      case Ohai['kernel']['machine']
-      when 'x86_64'
+      case `getconf LONG_BIT`
+      when /64\n?/
         'amd64'
-      when 'i686'
+      when /32\n?/
         'i386'
       when 'armv6l'
```

Comment:
Just as an fyi - this will totally break things that aren't x86 based. We should probably stick with `Ohai['kernel']['machine']